parser/src/lsp_hints.rs

```diff
@@ -0,0 +1,59 @@
+use crate::cfg::sexpr::{Span, Spanned};
+
```

Comment:
Can we put everything in the current file under a `mod lsp { ... }` and with cfg-lsp, do `pub use lsp::*;`?

Then for cfg-not-lsp, have `mod not_lsp { ... }` with the minimum number of struct definitions to allow the code to compile when the lsp feature is not enabled.

The thought here is to ensure that as much of the lsp-specific code as possible can be compiled out to help reduce code that dead code elimination needs to (and might for whatever reason fail to) remove, and this would also help remind and catch places where lsp code hasn't been marked with a cfg attribute.

Reply:
done in 0b1bf3db8435d3e0e53632168713ad320186dd1e